### PR TITLE
CSHARP-3535: Regression in support for $out to S3

### PR DESCRIPTION
**Pull request by rstam**

### CODE REVIEW

**Comment by rstam** [comment]
Evergreen patch:

https://evergreen.mongodb.com/version/606df637d6d80a48b7237b6b

File: tests/MongoDB.Driver.Core.Tests/Core/Operations/AggregateToCollectionOperationTests.cs [hunk 1]
```diff
@@ -70,6 +72,22 @@ public void Constructor_with_collectionNamespace_should_create_a_valid_instance(
             subject.WriteConcern.Should().BeNull();
         }
 
+        [Theory]
+        [InlineData("{ $out : 'collection' }", "{ $out : 'collection' }")]
+        [InlineData("{ $out : { db : 'database', coll : 'collection' } }", "{ $out : 'collection' }")]
+        [InlineData("{ $out : { db : 'differentdatabase', coll : 'collection' } }", "{ $out : { db : 'differentdatabase', coll : 'collection' } }")]
+        [InlineData("{ $out : { s3 : { } } }", "{ $out : { s3 : { } } }")]
```
Comment (rstam) [inline tests/MongoDB.Driver.Core.Tests/Core/Operations/AggregateToCollectionOperationTests.cs:79]: We didn't have any tests for $out to S3 so we didn't catch this regression

**Review by JamesKovacs — APPROVED** [review]
LGTM. I verified that the `s3`-related test fails without the fix and passes with it. Nice, straightforward, simple fix.

**Comment by rstam** [comment]
Squashed, rebased and pushed to master and backported to v2.12.x

**Pull request closed without merging**